### PR TITLE
Edit IAM permissions for new relic user

### PR DESCRIPTION
**Pull request by jzbahrai**
# Summary | Résumé

Task: [#49](https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/49)

Add permissions for new relic user with regard to the API lambda.

### CODE REVIEW

**Comment by ben851** [comment]
This looks good, although I'm concerned as to how this is currently working in both prod and staging without these permissions assigned to the policy... Unless there is a global policy that was done via clickops at some point. Lets try it in staging to see how it goes.

File: aws/lambda-api/iam.tf [hunk 1]
```diff
@@ -84,6 +84,17 @@ data "aws_iam_policy_document" "api_policies" {
...
+      "cloudwatch:ListMetrics",
+      "cloudwatch:GetMetricData",
+      "config:BatchGetResourceConfig",
+      "config:ListDiscoveredResources",
+      "tag:GetResources",
+      "lambda:GetAccountSettings",
+      "lambda:ListFunctions",
+      "lambda:ListAliases",
+      "lambda:ListTags",
+      "lambda:ListEventSourceMappings"
```
Comment (jimleroyer) [inline aws/lambda-api/iam.tf:97]: Is this the right place to add these? @ben851 indicated that the actions should be applied on `*` resources. Would the the new relic key resource need to do these actions? I might be misunderstanding something, iam is always some black magic to me.

Comment (ben851) [reply aws/lambda-api/iam.tf:97]: I think this is good, it's where I had them for mine. I am going off what was here:
https://docs.newrelic.com/docs/serverless-function-monitoring/aws-lambda-monitoring/enable-lambda-monitoring/account-linking/

New Relic's doc's are a bit confusing in that this is talking about linking aws accounts but underneath it's talking about using the lambda integration...

Comment (ben851) [reply aws/lambda-api/iam.tf:97]: Ok I blew it you're 100% right Jimmy. I missed that this was assigned to the secret arn only. I will request the change

**Review by ben851 — CHANGES_REQUESTED** [review]
As per my conversation with @jimleroyer, these permissions are not scoped to the secret manager ARN - they need to be scoped to all resources (*).  This can be done by simply creating a new block in data "aws_iam_policy_document" "api_policies"   adding these new values and setting the resources to *

**Review by ben851 — APPROVED** [review]
Looks good!

**Pull request merged**